papaparse.js

```diff
@@ -1866,24 +1866,25 @@ License: MIT
 					if (aborted)
 						break;
 				}
-				delete msg.results;	// free memory ASAP
 			}
 			else if (isFunction(worker.userChunk))
 			{
 				worker.userChunk(msg.results, handle, msg.file);
-				delete msg.results;
 			}
 		}
 
 		if (msg.finished && !aborted)
-			completeWorker(msg.workerId, msg.results);
+			completeWorker(msg.workerId, msg.results, msg.file);
+
+		delete msg.results;
 	}
 
-	function completeWorker(workerId, results) {
+	function completeWorker(workerId, results, file) {
 		var worker = workers[workerId];
-		if (isFunction(worker.userComplete))
-			worker.userComplete(results);
-		worker.terminate();
```

Comment:
Could you elaborate why wee need to terminate the worker only when a function is defined?

Reply:
I think I may have unintentionally changed the original flow while modifying the code. I'll revert back to. Thank you.